### PR TITLE
Created helper method

### PR DESCRIPTION
**Pull request by nimeshnayaju**
Added helper method 'insertProduct' to add a new product into the collection.

### CODE REVIEW

File: utils/controller.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const { StatusCode } = require('../shared/constants');
+
+/**
+ * Helper method to insert a Product into the collection
+ */
+exports.insertProduct = function(body, barcode, res) {
+  let newProduct = new models.Product ({
+    barcode: barcode || body.barcode,
+    name: body.product_name,
+    category: body.category.split('>').map(Function.prototype.call, String.prototype.trim),
```
Comment (ShekharDewan) [inline utils/controller.js]: Can we change the category structure as per our discussion for which we referenced this:

https://docs.google.com/document/d/1NLGdPTkV9savb8d1MPY2bYVERdfGwiGpd_PqaP-m3oI/edit

File: handlers/controller.js [hunk 2]
```diff
@@ -149,29 +113,12 @@ let addProductByLookup = async(req, res) => {
         // else
         // query barcodelookup for product
         let bclRes = await bc.lookup({key: process.env.BC_API_KEY, barcode: barcode});
+        console.log(bclRes);
         if(bclRes.statusCode !== StatusCode.OK) {
             console.error(`error looking up ${barcode} in barcodelookup`);
             return res.status(bclRes.statusCode).send({ data: bclRes.data });
         }
-
-        let newProduct = new models.Product ({
```
Comment (ShekharDewan) [inline handlers/controller.js]: Are we not doing this product structure anymore and straightaway adding data? I don't see the below default initializations anywhere.

File: dal/database.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require('dotenv').config();
 const mongoose = require('mongoose');
 const Product = require('../schemas/product');
-const uri = process.env.MONGODB_URI || '127.0.0.1';
+const uri = process.env.MONGODB_URI || 'mongodb://localhost:27017/test' || '127.0.0.1';
```
Comment (evscott) [inline dal/database.js]: We should only have two options here. Lets change `process.env.MONGODB_URI || 'mongodb://localhost:27017/test' || '127.0.0.1';` to `process.env.MONGODB_URI || 'mongodb://localhost:27017/test';`.

File: handlers/controller.js (hunk 2 above)
Comment (evscott) [reply handlers/controller.js]: Look at the newly added `Controller` file inside of `utils` @ShekharDewan.

**Pull request merged**